Command/PopulateCommand.php

```diff
@@ -109,20 +110,34 @@ protected function execute(InputInterface $input, OutputInterface $output)
      */
     private function populateIndex(OutputInterface $output, $index, $reset, $options)
     {
+
+        /** @var $providers ProviderInterface[] */
+        $providers = $this->providerRegistry->getIndexProviders($index);
+
         if ($reset) {
             $output->writeln(sprintf('<info>Resetting</info> <comment>%s</comment>', $index));
             $this->resetter->resetIndex($index, true);
         }
 
-        /** @var $providers ProviderInterface[] */
-        $providers = $this->providerRegistry->getIndexProviders($index);
-
         foreach ($providers as $type => $provider) {
-            $loggerClosure = function($message) use ($output, $index, $type) {
-                $output->writeln(sprintf('<info>Populating</info> %s/%s, %s', $index, $type, $message));
-            };
+            if (class_exists('Symfony\Component\Console\Helper\ProgressBar')) {
```

Comment:
Please move the creation of the loggerClosure to its own method.

I also think that instead of having the progress-bar option in the Provider we can just pass the necessary information as parameters to the loggerClosure and let it handle everything.

IE, we'd call `$loggerClosure($stepCount, $nbObjects, $stepStartTime)` and then format that information how we want it formatted depending on if ProgressBar is available.